ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc

```diff
@@ -894,62 +898,59 @@ int
 TFLiteCore::reloadModel (const char *_model_path)
 {
   int err;
+  TFLiteInterpreter * interpreter_temp = interpreter;
 
   if (!g_file_test (_model_path, G_FILE_TEST_IS_REGULAR)) {
     ml_loge ("The path of model file(s), %s, to reload is invalid.", _model_path);
     return -EINVAL;
   }
-
-  interpreter_sub.lock ();
-  interpreter_sub.setModelPath (_model_path);
+  interpreter_sub = new TFLiteInterpreter ();
+  interpreter_sub->setModelPath (_model_path);
 
   /**
    * load a model into sub interpreter. This loading overhead is indenendent
    * with main one's activities.
    */
-  err = interpreter_sub.loadModel (num_threads, delegate);
+  err = interpreter_sub->loadModel (num_threads, delegate);
   if (err != 0) {
     ml_loge ("Failed to load model %s\n", _model_path);
     goto out_unlock;
   }
-  err = interpreter_sub.setInputTensorProp ();
+  err = interpreter_sub->setInputTensorProp ();
   if (err != 0) {
     ml_loge ("Failed to initialize input tensor\n");
     goto out_unlock;
   }
-  err = interpreter_sub.setOutputTensorProp ();
+  err = interpreter_sub->setOutputTensorProp ();
   if (err != 0) {
     ml_loge ("Failed to initialize output tensor\n");
     goto out_unlock;
   }
-  err = interpreter_sub.cacheInOutTensorPtr ();
+  err = interpreter_sub->cacheInOutTensorPtr ();
   if (err != 0) {
     ml_loge ("Failed to cache input and output tensors storage\n");
     goto out_unlock;
   }
 
   /* Also, we need to check input/output tensors have the same info */
-  if (!gst_tensors_info_is_equal (interpreter.getInputTensorsInfo (),
-          interpreter_sub.getInputTensorsInfo ())
-      || !gst_tensors_info_is_equal (interpreter.getOutputTensorsInfo (),
-             interpreter_sub.getOutputTensorsInfo ())) {
+  if (!gst_tensors_info_is_equal (interpreter->getInputTensorsInfo (),
+          interpreter_sub->getInputTensorsInfo ())
+      || !gst_tensors_info_is_equal (interpreter->getOutputTensorsInfo (),
+             interpreter_sub->getOutputTensorsInfo ())) {
     ml_loge ("The model has unmatched tensors info\n");
     err = -EINVAL;
     goto out_unlock;
   }
 
-  /**
-   * Everything is ready. let's move the model in sub interpreter to main one.
-   * But, it needs to wait if main interpreter is busy (e.g., invoke()).
-   */
-  interpreter.lock ();
-  interpreter.moveInternals (interpreter_sub);
-  /* after this, all callbacks will handle operations for the reloaded model */
-  interpreter.unlock ();
+  interpreter_sub->lock ();
+  interpreter_temp->lock ();
 
-out_unlock:
-  interpreter_sub.unlock ();
+  interpreter = interpreter_sub;
 
+  interpreter_sub->unlock ();
+  interpreter_temp->unlock ();
+  delete interpreter_temp;
+out_unlock:
```

Comment:
How about removing this label and return directly?

Reply:
you're right. it was just the heritage of the previous code. it's removed.